lib/spack/spack/test/reporters.py

```diff
@@ -132,7 +132,9 @@ def test_reporters_extract_skipped(state):
     parts = spack.reporters.extract.extract_test_parts("fake", outputs)
 
     assert len(parts) == 1
-    parts[0]["completed"] == expected
+
+    # TODO: Fix this broken assertion
```

Comment:
@tldahlgren this test is broken, and it's not directly clear what it's doing.

Reply:
Yes. It is tied to `install_test.py`'s `test_process`.

There are only two cases at this point. We allow stand-alone testing of externally installed packages through `spack test run --external`. If they are not explicitly being tested, we flag and log that the test is skipped (e.g. `TestStatus.SKIPPED`). We will also log skipping if someone attempts to run tests on a spec that is not installed.

You can see where the report extraction grabs and flags the status in `reporters/extract.py`'s `extract_test_parts`.